src/components/vm/consoles/VncConsole.jsx

```diff
@@ -0,0 +1,155 @@
+import RFB_module from '@novnc/novnc/lib/rfb';
+const RFB = RFB_module.default;
+
+export const VncConsole = ({
+    children,
+    host,
+    port = '80',
+    path = '',
+    encrypt = false,
+    resizeSession = true,
+    clipViewport = false,
+    dragViewport = false,
+    scaleViewport = false,
+    viewOnly = false,
+    shared = false,
+    credentials,
+    repeaterID = '',
+    vncLogging = 'warn',
+    consoleContainerId,
+    onDisconnected = () => {},
+    onInitFailed,
+    onSecurityFailure,
+}) => {
+    const rfb = React.useRef();
+
+    const novncElem = React.useRef(null);
+
+    const _onDisconnected = React.useCallback(
+        (e) => {
+            onDisconnected(e);
+        },
+        [onDisconnected]
+    );
+
+    const _onSecurityFailure = React.useCallback(
+        (e) => {
+            onSecurityFailure(e);
+        },
+        [onSecurityFailure]
+    );
+
+    const addEventListeners = React.useCallback(() => {
+        if (rfb.current) {
+            rfb.current?.addEventListener('disconnect', _onDisconnected);
+            rfb.current?.addEventListener('securityfailure', _onSecurityFailure);
+        }
+    }, [rfb, _onDisconnected, _onSecurityFailure]);
+
+    const removeEventListeners = React.useCallback(() => {
+        if (rfb.current) {
+            rfb.current.removeEventListener('disconnect', _onDisconnected);
+            rfb.current.removeEventListener('securityfailure', _onSecurityFailure);
+        }
+    }, [rfb, _onDisconnected, _onSecurityFailure]);
+
+    const connect = React.useCallback(() => {
+        const protocol = encrypt ? 'wss' : 'ws';
+        const url = `${protocol}://${host}:${port}/${path}`;
+
+        const options = {
+            repeaterID,
+            shared,
+            credentials
+        };
+        rfb.current = new RFB(novncElem.current, url, options);
+        addEventListeners();
+        rfb.current.viewOnly = viewOnly;
+        rfb.current.clipViewport = clipViewport;
+        rfb.current.dragViewport = dragViewport;
+        rfb.current.scaleViewport = scaleViewport;
+        rfb.current.resizeSession = resizeSession;
+    }, [
+        addEventListeners,
+        host,
+        path,
+        port,
+        resizeSession,
+        clipViewport,
+        dragViewport,
+        scaleViewport,
+        viewOnly,
+        encrypt,
+        rfb,
+        repeaterID,
+        shared,
+        credentials
+    ]);
+
+    React.useEffect(() => {
+        initLogging(vncLogging);
+        try {
+            connect();
+        } catch (e) {
+            onInitFailed && onInitFailed(e);
+            rfb.current = undefined;
+        }
+
+        return () => {
+            disconnect();
+            removeEventListeners();
+            rfb.current = undefined;
+        };
+    }, [connect, onInitFailed, removeEventListeners, vncLogging]);
+
+    const disconnect = () => {
+        if (!rfb.current) {
+            return;
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2008-f1c9d954-20250715-172933-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)